staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -302,29 +301,45 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		// with non-equal specs (especially non-equal schemas).
 		var crds []*apiextensionsv1.CustomResourceDefinition
 		crds, err = r.crdLister.List(labels.Everything())
-		if err == nil {
-			if len(crds) == 0 {
-				err = errors.NewNotFound(schema.GroupResource{Group: apiextensionsv1.SchemeGroupVersion.Group, Resource: "customresourcedefinitions"}, "")
+		if err != nil {
+			utilruntime.HandleError(err)
+			responsewriters.ErrorNegotiated(
+				apierrors.NewInternalError(fmt.Errorf("error resolving resource")),
+				Codecs, schema.GroupVersion{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion}, w, req,
+			)
+			return
+		}
+
+		if len(crds) == 0 {
+			responsewriters.ErrorNegotiated(
```

Comment:
this is not correct. We have a delegate handler here (AFAIK). Fall back to that.